File: HotDesk_PRAI/TMS_HotDesk_Prai/App_Module/RangeValidator.ascx.vb

Partial Class App_Module_RangeValidator
    Inherits BaseUserControl
#Region "Variables"
    Private _dateType As String
#End Region
#Region "Properties"
    Public Property Text1() As String
        Get
            Return txtbox1.Text
        End Get
        Set(ByVal value As String)
            txtbox1.Text = value
        End Set
    End Property

    Public Property Text2() As String
        Get
            Return txtbox2.Text
        End Get
        Set(ByVal value As String)
            txtbox2.Text = value
        End Set
    End Property
    Public Property DataType() As ValidationDataType
        Get
            Return Me._dateType
        End Get
        Set(ByVal value As ValidationDataType)
            Me._dateType = value
        End Set
    End Property
#End Region
#Region "Methods"
    Public Sub BindSetting()
        REM assign validation group to multiple controls
        txtbox1.ValidationGroup = MyBase.ValidationGroup
        txtbox2.ValidationGroup = MyBase.ValidationGroup
        rfBox1.ValidationGroup = MyBase.ValidationGroup
        rfbox2.ValidationGroup = MyBase.ValidationGroup
        cvRange.ValidationGroup = MyBase.ValidationGroup
        cvRange.Type = Me.DataType
        cvCheckType1.ValidationGroup = MyBase.ValidationGroup
        cvCheckType1.Type = Me.DataType
        cvCheckType2.ValidationGroup = MyBase.ValidationGroup
        cvCheckType2.Type = Me.DataType

    End Sub
    Public Sub BindData()
        txtbox1.Text = txtbox1.Text
        txtbox2.Text = txtbox2.Text
    End Sub

#End Region
#Region "Events"
#Region "Overrides"
    Public Overrides Sub DataBind()
        Me.BindSetting()
        MyBase.DataBind()
    End Sub
#End Region
#End Region
End Class

File: TMS_HotDesk/App_Module/RangeValidator.ascx.cs
using System.Web.UI.WebControls;

public partial class App_Module_RangeValidator : BaseUserControl
{
    #region Variables
    private ValidationDataType _dateType;
    #endregion

    #region Properties
    public string Text1
    {
        get { return txtbox1.Text; }
        set { txtbox1.Text = value; }
    }

    public string Text2
    {
        get { return txtbox2.Text; }
        set { txtbox2.Text = value; }
    }

    public ValidationDataType DataType
    {
        get { return this._dateType; }
        set { this._dateType = value; }
    }
    #endregion

    public void BindSetting()
    {
        txtbox1.ValidationGroup = base.ValidationGroup;
        txtbox2.ValidationGroup = base.ValidationGroup;
        rfBox1.ValidationGroup = base.ValidationGroup;
        rfbox2.ValidationGroup = base.ValidationGroup;
        cvRange.ValidationGroup = base.ValidationGroup;
        cvRange.Type = this.DataType;
        cvCheckType1.ValidationGroup = base.ValidationGroup;
        cvCheckType1.Type = this.DataType;
        cvCheckType2.ValidationGroup = base.ValidationGroup;
        cvCheckType2.Type = this.DataType;
    }

    public void BindData()
    {
        txtbox1.Text = txtbox1.Text;
        txtbox2.Text = txtbox2.Text;
    }

    public override void DataBind()
    {
        this.BindSetting();
        base.DataBind();
    }
}
